integrations/microstream/health/src/main/java/io/helidon/integrations/microstream/health/MicrostreamHealthCheck.java

```diff
@@ -0,0 +1,177 @@
+
+            if (status.get()) {
+                builder.up();
+            } else {
+                builder.down();
+            }
+        } catch (Throwable e) {
+            builder.down();
+            builder.withData("ErrorMessage", e.getMessage());
+            builder.withData("ErrorClass", e.getClass().getName());
+        }
+
+        return builder.build();
+    }
+
+    /**
+     * Create a default health check for Microstream.
+     *
+     * @param embeddedStorageManager the EmbeddedStorageManager used by the the health check
+     * @return default health check for Microstream
+     */
+    public static MicrostreamHealthCheck create(EmbeddedStorageManager embeddedStorageManager) {
+        return builder(embeddedStorageManager).build();
+    }
+
+    /**
+     * A fluent API builder to create a health check for Microstream.
+     *
+     * @param embeddedStorageManager EmbeddedStorageManager
+     * @return a new builder
+     */
+    public static Builder builder(EmbeddedStorageManager embeddedStorageManager) {
+        return new Builder(embeddedStorageManager);
+    }
+
+    /**
+     *
+     * Builder for MicrostreamHealthCheck.
+     *
+     */
+    public static class Builder implements io.helidon.common.Builder<MicrostreamHealthCheck> {
+
+        private EmbeddedStorageManager embeddedStorageManager;
+        private long timeoutDuration;
+        private TimeUnit timeoutUnit;
+        private String name;
+
+        private Builder(EmbeddedStorageManager embeddedStorageManager) {
+            this.embeddedStorageManager = embeddedStorageManager;
+            this.name = DEFAULT_NAME;
+            this.timeoutDuration = DEFAULT_TIMEOUT_SECONDS;
+            this.timeoutUnit = TimeUnit.SECONDS;
+        }
+
+        @Override
+        public MicrostreamHealthCheck build() {
+            return new MicrostreamHealthCheck(this);
+        }
+
+        /**
+         * Customized name of the health check.
+         *
+         * @param name name of the health check
+         * @return updated builder instance
+         */
+        public Builder name(String name) {
+            this.name = name;
+            return this;
+        }
+
+        /**
+         * Set custom timeout to wait for statement execution response. Default value is
+         * 10 seconds.
+         *
+         * @param duration the maximum time to wait for statement execution response
+         * @param timeUnit the time unit of the timeout argument
+         * @return updated builder instance
+         */
+        public Builder timeout(long duration, TimeUnit timeUnit) {
+            this.timeoutDuration = duration;
+            this.timeoutUnit = timeUnit;
+            return this;
+        }
+
+        /**
+         * get the current set EmbeddedStorageManager.
+         *
+         * @return EmbeddedStorageManager
+         */
+        public EmbeddedStorageManager embeddedStorageManager() {

```

Comment:
There is no need for getters on a builder (unless the getter is used from another module). In general this increases the scope of public API.
You can use field access when within the same class.

Reply:
ok, getters removed